app/views/unattended/provisioning_templates/user_data/preseed_autoinstall_cloud_init.erb

```diff
@@ -19,10 +19,14 @@ autoinstall:
     geoip: false
     preserve_sources_list: false
     primary:
-      - arches: [amd64, i386]
-        uri: http://archive.ubuntu.com/ubuntu
-      - arches: [default]
-        uri: http://ports.ubuntu.com/ubuntu-ports
+      - arches: [amd64,i386]
+<% if host_param('ubuntu_mirror_uri') -%>
+        uri: <%= host_param('ubuntu_mirror_uri') %>
+<% else -%>
+        uri: http://<%= @preseed_server %><%= @preseed_path %>
```

Comment:
this breaks support for non-amd64, non-i386 installs.

Reply:
AFAIK it does not break support for non-amd64/i386 arch - there is no support in the first place. It removes the "official" repos and makes them dynamic or falls back to the Installation Media.

Do you suggest adding `@arch` here?

Still, since we disable all repo components [here](https://github.com/theforeman/foreman/pull/9363/files#diff-02b11c45a8fe92a6ae919a699d0198e01b0f9fca7b3449725e6406ebae836fa3R11), the repo should not be used at all since we want to prohibit automatic updates during the installer. The finish script should take care of this AFAIK.

Reply:
You are right - in a non-katello environment this is true. The content of `/etc/apt/sources.list` is kept since the finish script does not wipe it out. 
Still, the user has the option to use `ports.ubuntu.com` by passing it as the corresponding host param `ubuntu_mirror_uri` or as Installation Media.

For now, I didn't run it with any non-amd64 systems. But the arch is already hardcoded in the template and this PR doesn't change that. What about creating a follow-up PR which addresses the architecture?

Reply:
>  And if that system is not x86(_64) we used to use ports.ubuntu.com and now don't use anything, so the system won't have any repos enabled at all?

Just figured I didn't really answer this. I'm actually not aware of what a non-amd64 system does with such a configuration. I guess it is still going to have the same `sources.list` but just won't be able to use it.